txn.go

```diff
@@ -61,13 +62,34 @@ func (d *Dgraph) NewTxn() *Txn {
 	}
 }
 
+// NewTxnInNamespace creates a new transaction in a namespace.
+// If the namespace is non-existent, the transaction will run
+// in the default namespace. Note that a Tnx created with this function
+// against an ACL-enabled Dgraph cluster will not complete successfully as the
+// namespace from the JWT is used to calculate the namespace ID.
+func (d *Dgraph) NewTxnInNamespace(namespace string) *Txn {
+	txn := d.NewTxn()
+	txn.namespace = namespace
+	return txn
+}
+
 // NewReadOnlyTxn sets the txn to readonly transaction.
 func (d *Dgraph) NewReadOnlyTxn() *Txn {
 	txn := d.NewTxn()
 	txn.readOnly = true
 	return txn
 }
 
+// NewReadOnlyTxnInNamespace creates a new read-only transaction in a namespace.
+// If the namepace is non-existent, the transction will run in the default namespace.
+// Note that a Tnx created with this function against an ACL-enabled Dgraph cluster will not
+// complete successfully as the namespace from the JWT is used to calculate the namespace ID.
+func (d *Dgraph) NewReadOnlyTxnInNamespace(namespace string) *Txn {
```

Comment:
we should instead take an option, that's the pattern we have been using through the v2 APIs.